### PR TITLE
Improve usage of the term 'platform' in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Daniel-Khodabakhsh**
This PR improves `CONTRIBUTING.md` by replacing some instances of the term `Platform` with `GitHub` for the following reasons:

* This was unnecessary legalese which made the document harder to comprehend.
* Fixes a few ambiguous usages of the term:
  * In one usage there is ambiguity if there is a platform called "Platform".
  * Ambiguous with other usages of the word when referring to the runtime platform.
* Fixes one incorrect usage of the term, where reference to `"Platform"` was intended to refer to any platform OTHER THAN `GitHub` or `"the Platform"`.

---

- [x]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

### CODE REVIEW

File: CONTRIBUTING.md
Comment (chennes) [inline CONTRIBUTING.md:29]:
```suggestion
4. FreeCAD uses GitHub's issue-tracking system to track problems and contributions. For help requests and general discussions, use the project forum. 
```

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8653">pipeline status</a> for feature branch [PR_8653](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8653). Pipeline [789002546](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/789002546) was triggered at [e8e56f0](https://github.com/FreeCAD/FreeCAD/pull/8653/commits/e8e56f04839046e570181cf1bd4bbffc8de087ce). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Review by chennes — CHANGES_REQUESTED** [review]
Thanks for the PR -- one word change requested (see below).

File: CONTRIBUTING.md
Comment (Daniel-Khodabakhsh) [reply CONTRIBUTING.md]: Thanks for catching that!

**Review by chennes — APPROVED** [review]

**Comment by chennes** [comment]
I basically think this is a good idea: I'm holding off on merging immediately to give other Maintainers an opportunity to comment on the PR (since there isn't a GitHub Issue it's not clear that this is an "identified and agreed-on problem", to use the terms from the document).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8653">pipeline status</a> for feature branch [PR_8653](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8653). Pipeline [789005947](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/789005947) was triggered at [f443f27](https://github.com/FreeCAD/FreeCAD/pull/8653/commits/f443f2746d0750eede3b4ff422c6b478d4fec2ce). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by yorikvanhavre** [comment]
I fundamentally agree this should be an easy to understand document, although I'm a bit uneasy to add the name of a commercial product in our guidelines... But I have no better idea :/

**Comment by chennes** [comment]
@yorikvanhavre fundamentally that was the reason we used the "legalese" in the first place, to try to de-emphasize "GitHub". But it's probably not worth it if that causes confusion for new contributors. I'm going to merge this, and we can continue to refine the document over time if someone comes up with a better solution later on.

**Comment by yorikvanhavre** [comment]
Agree!

**Pull request merged**
